ompi/mca/coll/han/coll_han_component.c

```diff
@@ -177,18 +238,17 @@ static int han_register(void)
                                            MCA_BASE_VAR_SCOPE_READONLY, &cs->han_bcast_segsize);
 
     cs->han_bcast_up_module = 0;
-    (void) mca_base_component_var_register(c, "bcast_up_module",
-                                           "up level module for bcast, 0 libnbc, 1 adapt",
-                                           MCA_BASE_VAR_TYPE_INT, NULL, 0, 0,
-                                           OPAL_INFO_LVL_9,
-                                           MCA_BASE_VAR_SCOPE_READONLY, &cs->han_bcast_up_module);
+    (void) mca_coll_han_query_module_from_mca(c, "bcast_up_module",
+                                              "up level module for bcast, 0 libnbc, 1 adapt",
+                                              OPAL_INFO_LVL_9, &cs->han_bcast_up_module,
```

Comment:
My $0.02: if you're making these MCA vars, I'm kinda assuming you intend for end users to be able to tune this stuff, right?  If so, maybe make them level 6 (i.e., "Application tuner / all" per https://github.com/open-mpi/ompi/wiki/MCAParamLevels).

Reply:
All collective tuning MCA parameters are developer level (across all modules). If anybody volunteers to propose a patch to change them all in one go that's fine, but until them I would favor consistency and keep all collective tuning MCA on the same level.

Reply:
Ok.